### PR TITLE
Added C++ Utils for GMLImage

### PR DESCRIPTION
**Pull request by priankakariatyml**
1. Refactored GMLImage Utils to enable creation of C++ and C frame buffers with common functions.
2. Added C++ GMLImage Utils for creating C++ frame buffer from GMLImage.

### CODE REVIEW

File: tensorflow_lite_support/ios/task/vision/utils/sources/GMLImage+Utils.h [hunk 1]
```diff
@@ -53,13 +65,13 @@ NS_ASSUME_NONNULL_BEGIN
 /**
  * Loads an image from a file in an app bundle into a GMLImage object.
  *
- * @param classObject The specified class associated with the bundle containing
- * the file to be loaded.
+ * @param classObject The specified class associated with the bundle containingthe file to be
```
Comment (khanhlvg) [inline tensorflow_lite_support/ios/task/vision/utils/sources/GMLImage+Utils.h:68]: typo: containingthe

I think you're using a config with line width = 100 chars. Google's code require line width <= 80 chars so a lot of changes here are accidentally caused by that incorrect config.

File: tensorflow_lite_support/ios/task/vision/utils/sources/GMLImage+Utils.m [hunk 2]
```diff
@@ -265,17 +247,13 @@ + (UInt8 *_Nullable)pixelDataFromCGImage:(CGImageRef)cgImage error:(NSError **)e
...
-                                               buffer:buffer
-                                                error:error];
+  return buffer;
 }
 
-- (TfLiteFrameBuffer *)frameBufferFromCIImage:(CIImage *)ciImage error:(NSError **)error {
+- (uint8_t *)frameBufferFromCIImage:(CIImage *)ciImage error:(NSError **)error {
   uint8_t *buffer = NULL;
 
   int width = 0;
```
Comment (khanhlvg) [inline tensorflow_lite_support/ios/task/vision/utils/sources/GMLImage+Utils.m:259]: I think width and height aren't necessary here anymore.

File: tensorflow_lite_support/ios/task/vision/utils/sources/GMLImage+Utils.m [hunk 3]
```diff
@@ -193,6 +158,23 @@ - (TfLiteFrameBuffer *)frameBufferWithError:(NSError **)error {
...
+
+  if (self.CGImage) {
+    width = CGImageGetWidth(self.CGImage);
+    height = CGImageGetHeight(self.CGImage);
+  } else if (self.CIImage.pixelBuffer) {
+    width = CVPixelBufferGetWidth(self.CIImage.pixelBuffer);
+    height = CVPixelBufferGetHeight(self.CIImage.pixelBuffer);
+  } else if (self.CIImage.CGImage) {
+    width = CGImageGetWidth(self.CIImage.CGImage);
+    height = CGImageGetWidth(self.CIImage.CGImage);
```
Comment (khanhlvg) [inline tensorflow_lite_support/ios/task/vision/utils/sources/GMLImage+Utils.m:173]: CGImageGetHeight

**Review by khanhlvg — CHANGES_REQUESTED** [review]

File: tensorflow_lite_support/ios/task/vision/utils/sources/GMLImage+Utils.h (hunk 1 above)
Comment (priankakariatyml) [reply tensorflow_lite_support/ios/task/vision/utils/sources/GMLImage+Utils.h]: I have fixed the typo. Will the char length be resolved when you merge the PR?

File: tensorflow_lite_support/ios/task/vision/utils/sources/GMLImage+Utils.m (hunk 2 above)
Comment (priankakariatyml) [reply tensorflow_lite_support/ios/task/vision/utils/sources/GMLImage+Utils.m]: Fixed

File: tensorflow_lite_support/ios/task/vision/utils/sources/GMLImage+Utils.m (hunk 3 above)
Comment (priankakariatyml) [reply tensorflow_lite_support/ios/task/vision/utils/sources/GMLImage+Utils.m]: Fixed

File: tensorflow_lite_support/ios/task/vision/utils/sources/GMLImage+Utils.h (hunk 1 above)
Comment (priankakariatyml) [reply tensorflow_lite_support/ios/task/vision/utils/sources/GMLImage+Utils.h]: NVM I fixed the line width.

File: tensorflow_lite_support/ios/task/vision/utils/sources/GMLImage+Utils.m (hunk 3 above)
Comment (priankakariatyml) [reply tensorflow_lite_support/ios/task/vision/utils/sources/GMLImage+Utils.m]: Fixed

File: tensorflow_lite_support/ios/task/vision/utils/sources/GMLImage+Utils.h [hunk 4]
```diff
@@ -24,26 +25,39 @@ NS_ASSUME_NONNULL_BEGIN
...
+ * saved. If `nil`, no error will be saved.
+ *
+ * @return The TfLiteFrameBuffer created from the gmlImage which can be used 
  * with the TF Lite Task Vision C library.
  */
 - (nullable TfLiteFrameBuffer *)cFrameBufferWithError:(NSError *_Nullable *)error;
 
 /**
- * Gets grayscale pixel buffer from GMLImage if source type is
+ * Gets grayscale pixel buffer from GMLImage if source type is
```
Comment (schmidt-sebastian) [inline tensorflow_lite_support/ios/task/vision/utils/sources/GMLImage+Utils.h:56]: Extra whitespace (will fix on import)

**Review by schmidt-sebastian — APPROVED** [review]

**Pull request merged**